### PR TITLE
[Doc] Update *Develop Connectors Guide*

### PR DESCRIPTION
**Pull request by Anonymitaet**

### CODE REVIEW

File: site2/docs/io-develop.md [hunk 1]
```diff
@@ -1,156 +1,188 @@
...
 
-#### Develop a sink connector
+    * {@inject: github:`Record`:/pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/Record.java#L28} should provide the following variables:
 
-Developing a sink connector is as easy as developing a source connector. You just need to
-implement {@inject: github:`Sink`:/pulsar-io/core/src/main/java/org/apache/pulsar/io/core/Sink.java} interface.
+      |Variable|Required|Description
+      |---|---|---
+      `TopicName`|No|Pulsar topic name from which the record is originated from.
+      `Key`|No| |
```
Comment (Anonymitaet) [inline site2/docs/io-develop.md]: @tuteng It seems that the description of `Key` in [`Record.java`](https://github.com/apache/pulsar/blob/master/pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/Record.java#L28) is incorrect.

Could you please provide a description for `Key`?

File: site2/docs/io-develop.md [hunk 2]
```diff
@@ -1,156 +1,188 @@
...
+      `Value`|Yes|Actual data of the record.
+      `EventTime`|No|Event time of the record from the source.
+      `PartitionId`|No| If the record is originated from a partitioned source, it returns its `PartitionId`. <br/><br/>`PartitionId` is used as a part of the unique identifier by Pulsar IO runtime to deduplicate messages and achieve exactly-once processing guarantee.
+      `RecordSequence`|No|If the record is originated from a sequential source, it returns its `RecordSequence`.<br/><br/>`RecordSequence` is used as a part of the unique identifier by Pulsar IO runtime to deduplicate messages and achieve exactly-once processing guarantee.
+      `Properties` |Yes| If the record carries user-defined properties, it returns those properties.
+      `DestinationTopic`|No|Topic to which message should be written.
+      `Message`|No||
```
Comment (Anonymitaet) [inline site2/docs/io-develop.md]: @tuteng The description of `Message` is missing in [`Record.java`](https://github.com/apache/pulsar/blob/master/pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/Record.java#L28).

Could you please provide a description for `Message`?

File: site2/docs/io-develop.md (hunk 1 above)
Comment (tuteng) [reply site2/docs/io-develop.md]: `Key` is related to the routing mode of the message http://pulsar.apache.org/docs/en/concepts-messaging/#routing-modes

File: site2/docs/io-develop.md [hunk 3]
```diff
@@ -1,156 +1,188 @@
...
-implement {@inject: github:`Sink`:/pulsar-io/core/src/main/java/org/apache/pulsar/io/core/Sink.java} interface.
+      |Variable|Required|Description
+      |---|---|---
+      `TopicName`|No|Pulsar topic name from which the record is originated from.
+      `Key`|No| |
+      `Value`|Yes|Actual data of the record.
+      `EventTime`|No|Event time of the record from the source.
+      `PartitionId`|No| If the record is originated from a partitioned source, it returns its `PartitionId`. <br/><br/>`PartitionId` is used as a part of the unique identifier by Pulsar IO runtime to deduplicate messages and achieve exactly-once processing guarantee.
+      `RecordSequence`|No|If the record is originated from a sequential source, it returns its `RecordSequence`.<br/><br/>`RecordSequence` is used as a part of the unique identifier by Pulsar IO runtime to deduplicate messages and achieve exactly-once processing guarantee.
+      `Properties` |Yes| If the record carries user-defined properties, it returns those properties.
```
Comment (tuteng) [inline site2/docs/io-develop.md]: `Properties ` this option also seems to be optional.

**Comment by Anonymitaet** [comment]
@tuteng I've incorporated all your comments, could u PTAL?

**Comment by Anonymitaet** [comment]
run java8 tests

**Comment by Anonymitaet** [comment]
run java8 tests

**Comment by Anonymitaet** [comment]
run java8 tests

**Pull request merged**